src/common/exponent/exponentHelper.ts

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as path from "path";
+import * as Q from "q";
+import * as XDL from "xdl";
+
+import {FileSystem} from "../node/fileSystem";
+import {Package} from "../node/package";
+import {ReactNativeProjectHelper} from "../reactNativeProjectHelper";
+import {CommandExecutor} from "../commandExecutor";
+import {Log} from "../log/log";
+
+const VSCODE_EXPONENT_JSON = "vscodeExponent.json";
+const EXPONENT_ENTRYPOINT = "exponentEntrypoint.js";
+const EXPONENT_INDEX = "exponentIndex.js";
+const DEFAULT_IOS_INDEX = "index.ios.js";
+const DEFAULT_ANDROID_INDEX = "index.android.js";
+const EXP_JSON = "exp.json";
+
+enum ReactNativePackageStatus {
+    FACEBOOK_PACKAGE,
+    EXPONENT_PACKAGE,
+    UNKNOWN
+}
+
+export class ExponentHelper {
+    private rootPath: string;
+    private fileSystem: FileSystem;
+    private commandExecutor: CommandExecutor;
+
+    private expSdkVersion: string;
+    private entrypointFilename: string;
+    private entrypointComponentName: string;
+
+    private dependencyPackage: ReactNativePackageStatus;
+
+    public constructor(projectRootPath: string) {
+        this.rootPath = projectRootPath;
+        this.fileSystem = new FileSystem();
+        this.commandExecutor = new CommandExecutor(this.rootPath);
+        this.dependencyPackage = ReactNativePackageStatus.UNKNOWN;
+        XDL.Config.validation.reactNativeVersionWarnings = false;
+    }
+
+    /**
+     * Create exp.json file in the workspace root
+     */
+    public createExpJson(): Q.Promise<void> {
+        let defaultSettings = {
+            "sdkVersion": "",
+            "entryPoint": EXPONENT_INDEX,
+            "slug": "",
+        };
+        return this.readVscodeExponentSettingFile()
```

Comment:
I was thinking on this exact same thing, and that's why I decided to add the `createOrOverrideExpJson` to `.vscode/vscodeExponent.json` (which defaults to true). 

We want to create a new `exp.json` everytime, because if the user updates the react-native version or the package name (in exponent words `slug`), we want to be able to update `exp.json`. But if the users needs to have their customized version of `exp.json`, then they can just change `createOrOverrideExpJson` to false and we will not create or overwrite `exp.json`